### ISSUE TITLE
Larger max size for README and CHANGELOG

### ISSUE THREAD

**Issue by rydmike** [dart-lang/pub#3543]
# Max size of README and CHANGELOG too small

The maximum char size (128kB) I assume, for README and CHANGELOG in pub.dev packages is too small.

Can you please increase the size?

Shame to have to delete history from CHANGELOG, which I now have to do for the change log in this case. OK so I like to be verbose, I don't do terse 😄 

I could move the CHANGELOG history to an offsite link, and only offer link to content for every version in the log. Already had to move the docs to its own site.

```
Publishing is forever; packages cannot be unpublished.
Policy details are available at https://pub.dev/policy

Do you want to publish flex_color_scheme 6.0.0-dev.1 to https://pub.dartlang.org (y/N)? y
Uploading...
`CHANGELOG.md` exceeds the maximum content length (131072 bytes).
pub finished with exit code 1

```

**Comment by sigurdm** [comment]
I'm fine with upping the limit to 512k or whatever as long as we have a limit in place.

@jonasfj and @isoos do you see any problems here?

**Comment by jonasfj** [comment]
side note: @rydmike you have an awesome [changelog](https://github.com/rydmike/flex_color_scheme/blob/e3e83add3252414890bb8f2a79e106bc197450e5/CHANGELOG.md) :rocket: 

We absolute should encourage writing great changelogs.

I think datastore will limit us to [1MB](https://cloud.google.com/datastore/docs/concepts/limits#limits) (so long as we keep it unindexed), and we probably have to leave room for a few other attributes.

So, yeah, bumping the limit to 512k seems reasonably, but we should do a test case on staging to make sure it works.


--------------------

I think we only did this because wanted to set some limit to prevent degenerate changelog files.
And ofcourse there is some point at which the solution is to put a link a the bottom saying that for versions prior to 1.2.3, see `CHANGELOG-pre-1.2.3.md` or something like that.

**Comment by isoos** [comment]
I am 50-50% on this, maybe slightly to concur: I think we should rather keep the current limit.

One part is Datastore: besides the single entity limit, we also have an overall transaction limit, and at upload time we write a lot of entries in a single transaction. (We could separate the content extraction from the main transaction though, but that introduces an additional backfill process).

The other part is the sheer size: the current changelog for `flex_color_scheme` is 40+ pages. It is impressively detailed, but I doubt that users of v5.1 are interested in changes that happened 1.5+ years ago in v1.3.

I think we should figure out how these files could grow and how they should be displayed. Maybe we need to customize the content extraction of the changelog, and display only the last-N versions (or up until 128 Kb). That way the input file could keep its size, and we also limit the size to the common query patterns.

**Comment by jonasfj** [comment]
> The other part is the sheer size: the current changelog for flex_color_scheme is 40+ pages.

Yeah, it's not clear that users would suffer from having to follow a link to read the CHANGELOG for very old versions.
Or we could support something like `git` does with a file per release:
https://github.com/git/git/tree/master/Documentation/RelNotes
